### PR TITLE
Convert absolute URL of anchor to relative

### PR DESCRIPTION
**Pull request by thomsebastin**
JIRA: https://jira.corp.adobe.com/browse/EXLM-269

### CODE REVIEW

File: src/modules/utils/dom-utils.js [hunk 1]
```diff
@@ -293,3 +293,61 @@ export const groupWithParagraphs = (document, nodes) => {
...
+}
+
+/**
+ * Handles converting absolute URLs to relative URLs for links and images within a document.
+ *
+ * @param {Document} document - The HTML document to process.
+ */
+export const handleAbsoluteUrl = (document) => {
+  updateLink(document, 'a', 'href');
+  updateLink(document, 'img', 'src');
```
Comment (Sivaramvt) [inline src/modules/utils/dom-utils.js:352]: Wouldn't this on  break all images in the site as we are not serving the images yet in custom domain @thomsebastin ?

Comment (thomsebastin) [reply src/modules/utils/dom-utils.js]: @Sivaramvt That makes sense. I have excluded image urls from the current implementation.

File: src/modules/utils/dom-utils.js [hunk 2]
```diff
@@ -293,3 +293,60 @@ export const groupWithParagraphs = (document, nodes) => {
...
+ * @param {string} selector - A CSS selector for selecting elements to update.
+ * @param {string} attribute - The attribute to update (e.g., 'href' or 'src').
+ */
+function updateLink(document, selector, attribute) {
+  const elements = document.querySelectorAll(selector);
+  if (!elements) return;
+
+  elements.forEach((el) => {
+    const url = el.getAttribute(attribute);
+    const baseUrl = 'https://experienceleague.adobe.com';
```
Comment (ahmed-musallam) [inline src/modules/utils/dom-utils.js:339]: There maybe other domains we need to do this, but good start here.

Can we move all of these utilities to a new file, call it `link-utils` ? (this dom util file is getting large)

**Review by ahmed-musallam — CHANGES_REQUESTED** [review]

File: src/modules/utils/dom-utils.js (hunk 2 above)
Comment (thomsebastin) [reply src/modules/utils/dom-utils.js]: @ahmed-musallam I have moved to `link-utils`.

**Review by Sivaramvt — APPROVED** [review]

**Pull request merged**